reconstruction/ecoli/dataclasses/process/external_pathway.py

```diff
@@ -0,0 +1,264 @@
+
+
+    def __getstate__(self):
+        """Return the state to pickle, omitting derived attributes that
+        __setstate__() will recompute, esp. the ode_derivatives
+        that don't pickle.
+        """
+        try:
+            return data.dissoc_strict(self.__dict__, (
+                'symbolic_rates',
+			    '_rates',
+			    '_stoich_matrix'))
+        except:
+            return self.__dict__
+
+
+    def __setstate__(self, state):
+        """Restore instance attributes, recomputing some of them."""
+        try:
+            self.__dict__.update(state)
+            self._populate_derivative_and_jacobian()
+        except:
+            return 0
+
+    def stoich_matrix(self):
+        '''
+        Builds stoichiometry matrix
+        Rows: molecules
+        Columns: reactions
+        Values: reaction stoichiometry
+        '''
+        shape = (self._stoichMatrixI.max() + 1, self._stoichMatrixJ.max() + 1)
+        out = np.zeros(shape, np.float64)
+        out[self._stoichMatrixI, self._stoichMatrixJ] = self._stoichMatrixV
+        return out
+
+    def mass_matrix(self):
+        '''
+        Builds stoichiometry mass matrix
+        Rows: molecules
+        Columns: reactions
+        Values: molecular mass
+        '''
+        shape = (self._stoichMatrixI.max() + 1, self._stoichMatrixJ.max() + 1)
+        out = np.zeros(shape, np.float64)
+        out[self._stoichMatrixI, self._stoichMatrixJ] = self._stoich_matrix_mass
+        return out
+
+    def mass_balance(self):
+        '''
+        Sum along the columns of the massBalance matrix to check for reaction mass balance
+        '''
+        return np.sum(self.balance_matrix, axis=0)
+
+    def _populate_derivative_and_jacobian(self):
+        '''Compile callable functions for computing the derivative.'''
+        self._make_derivative()
+
+        self._rates = build_ode.derivatives(self.symbolic_rates)
+        self._stoich_matrix = self.stoich_matrix()  # Matrix is small and can be cached for derivatives
+
+
+    def _make_y_dy(self):
+        '''This is building the right hand side of the ODE system using MM kinetics.'''
+
+        substrates = self.substrates
+        yStrings = ["y[%d]" % x for x in range(len(substrates))]
+        y = sp.symbols(yStrings)
+        rates = []
+        for colIdx in range(len(substrates)):
+            flux = self.kcats[colIdx] * y[colIdx]/ (self.kms[colIdx] + y[colIdx])
+            rates.append(flux)
+        return y, rates
+
+    def _make_derivative(self):
+        '''
+        Creates symbolic representation of the ordinary differential equations
+        and the Jacobian. Used during simulations.
+        '''
+        y, rates = self._make_y_dy()
+
+        rates = sp.Matrix(rates)
+        self.symbolic_rates = rates
+
+
+    def molecules_to_next_time_step(self, moleculeDict, enzymeDict, cellVolume,
+                                    nAvogadro, timeStepSec, method="LSODA", jit=True):
```

Comment:
`jit` is not being used anywhere inside this function, so this could be removed.